ddtrace/datadog/src/exporter.cpp

```diff
@@ -0,0 +1,500 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+// High-level skip for invalid architectures
+#ifndef __linux__
+#elif __aarch64__
+#elif __i386__
+#else
+
+#include "exporter.hpp"
+
+#include <iostream>
+#include <thread>
+
+using namespace Datadog;
+
+inline ddog_CharSlice to_slice(const std::string_view &str) {
+  return {.ptr = str.data(), .len = str.size()};
+}
+
+DdogProfExporter::DdogProfExporter(std::string_view env,
+                                   std::string_view service,
+                                   std::string_view version, 
+                                   std::string_view url) {
+
+  // Setup the 
+  ddog_Vec_Tag tags = ddog_Vec_Tag_new();
+  add_tag(tags, "language", language);
+  add_tag(tags, "env", env);
+  add_tag(tags, "service", service);
+  add_tag(tags, "version", version);
+  add_tag(tags, "profiler_version", profiler_version);
+
+  ddog_prof_Exporter_NewResult new_exporter = ddog_prof_Exporter_new(
+      to_slice("dd-trace-py"),
+      to_slice(profiler_version),
+      to_slice(family),
+      &tags,
+      ddog_Endpoint_agent(to_slice(url)));
+  ddog_Vec_Tag_drop(tags);
+
+  if (new_exporter.tag == DDOG_PROF_EXPORTER_NEW_RESULT_OK)
+    ptr = new_exporter.ok;
+  else
+    std::cout << "ERROR INITIALIZING LIBDATADOG EXPORTER" << std::endl;
+}
+
+DdogProfExporter::~DdogProfExporter() {
+  ddog_prof_Exporter_drop(ptr);
```

Comment:
we could make this a unique ptr with a custom deleter (I'm mostly thinking of how to remove the copy constructors as I'm not sure how this class is used)

Reply:
> not sure how this class is used

`DdogProfExporter` is an opaque wrapper class over libdatadog's `ddog_prof_Exporter`.   Every instance of `ddog_prof_Exporter` is under the management of libdatadog, so they are returned as pointers.  I wanted to consolidate the C-isms of the class into RAII, so

1. `DdogProfExporter` calls the appropriate libdatadog functions to create and destroy the `ddog_prof_Exporter` instance.  It also includes a few string-invariants which are only sent to libdatadog during instantiation of `ddog_prof_Exporter`.
2. the `Uploader` class takes a unique_ptr to `DdogProfExporter`.  This is mostly done to defer initialization of the field, so it occurs once inside of the constructor.

The lifetime and scope of `Uploader` and `DdogProfExporter` are comparable, so splitting the wrapper into its own class is a question of aesthetics, as far as I can tell.